include/volume/volume_cooling_gaussians.hpp

```diff
@@ -204,56 +204,53 @@ void compute_annealing_schedule(Polytope& P,
     get_first_gaussian(P, frac, chebychev_radius, error, a_vals);
 
 #ifdef VOLESTI_DEBUG
-    std::cout<<"first gaussian computed\n"<<std::endl;
+    std::cout << "first gaussian computed\n" << std::endl;
 #endif
 
     NT a_stop = 0.0;
     const NT tol = 0.001;
     unsigned int it = 0;
     unsigned int n = P.dimension();
-    const unsigned int totalSteps = ((int)150/((1.0 - frac) * error))+1;
-
-    if (a_vals[0]<a_stop) a_vals[0] = a_stop;
-
-#ifdef VOLESTI_DEBUG
-    std::cout<<"Computing the sequence of gaussians..\n"<<std::endl;
-#endif
+    const unsigned int totalSteps = static_cast<unsigned int>(150 / ((1.0 - frac) * error)) + 1;
 
     Point p(n);
 
     while (true)
     {
         // Compute the next gaussian
         NT next_a = get_next_gaussian<RandomPointGenerator>
-                      (P, p, a_vals[it], N, ratio, C, walk_length, rng);
+                    (P, p, a_vals[it], N, ratio, C, walk_length, rng);
+
+#ifdef VOLESTI_DEBUG
+        std::cout << "Processing a_vals[" << it << "] = " << a_vals[it] << ", next_a = " << next_a << std::endl;
+#endif
 
         NT curr_fn = 0;
         NT curr_its = 0;
-        auto steps = totalSteps;
 
-        WalkType walk(P, p, a_vals[it], rng);
-        //TODO: test update delta here?
+        if constexpr (std::is_same<WalkTypePolicy, GaussianHamiltonianMonteCarloExactWalk>::value) {
+            GaussianHamiltonianMonteCarloExactWalk::Walk<Polytope, RandomNumberGenerator> walk(P, p, a_vals[it], rng, walk_params);
 
-        update_delta<WalkType>
-                ::apply(walk, 4.0 * chebychev_radius
-                        / std::sqrt(std::max(NT(1.0), a_vals[it]) * NT(n)));
+            for (unsigned int j = 0; j < totalSteps; ++j) {
+                walk.template apply(P, p, a_vals[it], walk_length, rng);
+                curr_its++;
+                curr_fn += eval_exp(p, next_a) / eval_exp(p, a_vals[it]);
+            }
+        } else {
+            WalkTypePolicy walk(P, p, a_vals[it], rng);
 
-        // Compute some ratios to decide if this is the last gaussian
-        for (unsigned  int j = 0; j < steps; j++)
-        {
-            walk.template apply(P, p, a_vals[it], walk_length, rng);
-            curr_its += 1.0;
-            curr_fn += eval_exp(p, next_a) / eval_exp(p, a_vals[it]);
+            for (unsigned int j = 0; j < totalSteps; ++j) {
+                walk.template apply(P, p, a_vals[it], walk_length, rng);
+                curr_its++;
```

Comment:
this not an integer, although I do not know the reason (@TolisChal ?)

Reply:
Yes, that could be an integer